### PR TITLE
DSD-524: Chakra Notification

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-524](https://jira.nypl.org/browse/DSD-524)

## This PR does the following:
- Updates the `Notification` component to use Chakra.
- Note: I removed the export for the `NotificationContent` and `NotificationHeading` components and they are now props instead.

## How has this been tested?

Storybook.

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] View [the example in Storybook]()

### CODE REVIEW

File: src/theme/components/notification.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  baseStyle: ({ centered, notificationType }) => {
+    let color = "ui.black";
+    if (notificationType === NotificationTypes.Announcement) {
+      color = "section.research.secondary";
+    } else if (notificationType === NotificationTypes.Warning) {
+      color = "brand.primary";
+    }
+    return {
+      display: "flex",
+      marginBottom: "s",
```
Comment (bigfishdesign13) [inline src/theme/components/notification.ts]: Change this to `xxs`.  As I look at this in the browser, I am not sure why this margin was originally set as `s`.  That's too large a space.

File: src/components/Button/Button.tsx [hunk 2]
```diff
@@ -26,6 +26,8 @@ interface ButtonProps {
   mouseDown?: boolean;
   /** The action to perform on the `<button>`'s onClick function */
   onClick?: (event: React.MouseEvent | React.KeyboardEvent) => void;
+  /** Optionally pass in additional Chakra-based styles. */
+  sx?: { [key: string]: any };
```
Comment (bigfishdesign13) [inline src/components/Button/Button.tsx]: Why were these changes included in this PR?

File: src/components/Notification/Notification.tsx [hunk 3]
```diff
@@ -1,201 +1,183 @@
...
   id?: string;
-  /** Optional prop to control the margin around the `Notification` component */
+  /** Optional prop to control the margin around the `Notification` component. */
   noMargin?: boolean;
-  /** Optional prop to control the coloring of notification text and the visibility of an applicable icon */
-  notificationType?: NotificationTypes;
-  /** Modifiers array for use with BEM. See how to work with modifiers and BEM here: http://getbem.com/introduction/ */
-  modifiers?: string[];
+  /** Content to be rendered in a `NotificationContent` component. */
+  notificationContent: JSX.Element;
```
Comment (bigfishdesign13) [inline src/components/Notification/Notification.tsx]: Why `JSX.Element` for this and `notificationHeading`?  Why not define them as `string`?

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
Another complicated conversion.  Overall, it works well.  I have two questions and one requested change.

File: src/components/Notification/Notification.tsx (hunk 3 above)
Comment (EdwinGuzman) [reply src/components/Notification/Notification.tsx]: Right, I wasn't sure what the right type would be and maybe it can be `string | JSX.Element`.  All the previous examples were plain strings but what if more complicated (such as <b>, <i>, or links) elements should be rendered?

File: src/components/Button/Button.tsx (hunk 2 above)
Comment (EdwinGuzman) [reply src/components/Button/Button.tsx]: Because the `Button` component inside the `Notification` component has updated styles and I'm passing those through this prop - https://github.com/NYPL/nypl-design-system/pull/724/files#diff-1019e818df12ccb2db0eafad4b181840c4eeecd95b057fe177466b21a68d7b54R129

But, as mentioned in an earlier PR, the "sx" name is not the best so I'll update this.

Comment (bigfishdesign13) [reply src/components/Button/Button.tsx]: OK, got it.

File: src/components/Notification/Notification.tsx (hunk 3 above)
Comment (bigfishdesign13) [reply src/components/Notification/Notification.tsx]: Right, `notificationContent` might have more sophisiticated content, so allowing for `JSX.Element` does make sense.  But I think `notificationHeading` can just be `string`.

**Comment by EdwinGuzman** [comment]
Updated.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr724-u1ernhaulbukwkf8kdly8hhkuxytjbpd.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/616891f11ff56a6b8be90167

File: src/components/Notification/Notification.tsx (hunk 3 above)
Comment (bigfishdesign13) [reply src/components/Notification/Notification.tsx]: I see that `Hero` uses the `string | JSX.Element` pattern, so let's do the same here.  More often than not, the actual content for this section is going to be a plain string.

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
Let's go ahead and change the assignment of `notificationContent` to `string | JSX.Element` and bring it into alignment with what had been done in `Hero`.

**Comment by EdwinGuzman** [comment]
Updated and also updated the `sx` prop name to `additionalStyles`. I think that makes it clearer that it's a prop for the DS components whereas the Chakra components use `__css` or `sx`.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr724-u1ernhaulbukwkf8kdly8hhkuxytjbpd.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/616891f11ff56a6b8be90167

**Review by bigfishdesign13 — APPROVED** [review]
Very nice.
